File: solutions/1280-diet-plan-performance/diet-plan-performance.ts
function dietPlanPerformance(
    calories: number[],
    k: number,
    lower: number,
    upper: number,
): number {
    let score = 0;
    let sum = 0;
    for (let i = 0; i < k; i++) {
        sum += calories[i];
    }
    for (let i = k; i < calories.length; i++) {
        if (sum < lower) {
            score--;
        } else if (sum > upper) {
            score++;
        }
        sum += calories[i] - calories[i - k];
    }
    if (sum < lower) {
        score--;
    } else if (sum > upper) {
        score++;
    }
    return score;
}

File: solutions/1280-diet-plan-performance/diet-plan-performance.go
func dietPlanPerformance(calories []int, k int, lower int, upper int) int {
	sum, score := 0, 0
	for i := 0; i < k; i++ {
		sum += calories[i]
	}
	for i := k; ; i++ {
		if sum < lower {
			score--
		} else if sum > upper {
			score++
		}
		if i == len(calories) {
			break
		}
		sum += calories[i] - calories[i-k]
	}
	return score
}